pkg/resources/account.go

```diff
@@ -2,403 +2,481 @@ package resources
+		d.Set("is_org_admin", booleanStringFromBool(*account.IsOrgAdmin)),
+	); err != nil {
+		return nil, err
+	}
+
+	return []*schema.ResourceData{d}, nil
+}
+
+func CreateAccount(ctx context.Context, d *schema.ResourceData, meta any) diag.Diagnostics {
 	client := meta.(*provider.Context).Client
 
-	name := d.Get("name").(string)
-	objectIdentifier := sdk.NewAccountObjectIdentifier(name)
+	isOrgAdmin, err := client.ContextFunctions.IsRoleInSession(ctx, snowflakeroles.Orgadmin)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	if !isOrgAdmin {
+		return diag.FromErr(errors.New("current user doesn't have the orgadmin role in session"))
+	}
+
+	id := sdk.NewAccountObjectIdentifier(d.Get("name").(string))
 
-	createOptions := &sdk.CreateAccountOptions{
+	opts := &sdk.CreateAccountOptions{
 		AdminName: d.Get("admin_name").(string),
 		Email:     d.Get("email").(string),
 		Edition:   sdk.AccountEdition(d.Get("edition").(string)),
 	}
 
-	// get optional fields.
 	if v, ok := d.GetOk("admin_password"); ok {
-		createOptions.AdminPassword = sdk.String(v.(string))
+		opts.AdminPassword = sdk.String(v.(string))
 	}
 	if v, ok := d.GetOk("admin_rsa_public_key"); ok {
-		createOptions.AdminRSAPublicKey = sdk.String(v.(string))
+		opts.AdminRSAPublicKey = sdk.String(v.(string))
+	}
+	if v, ok := d.GetOk("admin_user_type"); ok {
+		userType, err := sdk.ToUserType(v.(string))
+		if err != nil {
+			return diag.FromErr(err)
+		}
+		opts.AdminUserType = &userType
 	}
 	if v, ok := d.GetOk("first_name"); ok {
-		createOptions.FirstName = sdk.String(v.(string))
+		opts.FirstName = sdk.String(v.(string))
 	}
 	if v, ok := d.GetOk("last_name"); ok {
-		createOptions.LastName = sdk.String(v.(string))
+		opts.LastName = sdk.String(v.(string))
 	}
-
-	// Has default, don't fetch with GetOk because this can be falsey and valid
-	v := d.Get("must_change_password")
-	createOptions.MustChangePassword = sdk.Bool(v.(bool))
-
-	if v, ok := d.GetOk("region_group"); ok {
-		createOptions.RegionGroup = sdk.String(v.(string))
-	} else {
-		// For organizations that have accounts in multiple region groups, returns <region_group>.<region> so we need to split on "."
-		currentRegion, err := client.ContextFunctions.CurrentRegion(ctx)
+	if v := d.Get("must_change_password"); v != BooleanDefault {
+		parsedBool, err := booleanStringToBool(v.(string))
 		if err != nil {
 			return diag.FromErr(err)
 		}
-		regionParts := strings.Split(currentRegion, ".")
-		if len(regionParts) == 2 {
-			createOptions.RegionGroup = sdk.String(regionParts[0])
-		}
+		opts.MustChangePassword = &parsedBool
+	}
+	if v, ok := d.GetOk("region_group"); ok {
+		opts.RegionGroup = sdk.String(v.(string))
 	}
 	if v, ok := d.GetOk("region"); ok {
-		createOptions.Region = sdk.String(v.(string))
-	} else {
-		// For organizations that have accounts in multiple region groups, returns <region_group>.<region> so we need to split on "."
-		currentRegion, err := client.ContextFunctions.CurrentRegion(ctx)
```

Comment:
This logic doesn't make sense (at least now), because when nothing is set explicitly the current region group is used (which is basically what this code did). The behavior didn't change afaik.